include/jlcxx/module.hpp

```diff
@@ -1188,23 +1188,21 @@ TypeWrapper<T> Module::add_type_internal(const std::string& name, JLSuperT* supe
   protect_from_gc(box_dt);
 
   // Register the type
-  if(!is_parametric)
+  if(is_parametric)
+  {
+    set_const(name, base_dt->name->wrapper);
```

Comment:
Using `static_cast<jl_value_t*>()` around for the argument here and in the next line fixes this. The reason is that otherwise a `jl_value_t*&` is passed.

Reply:
OK, can do, but why did it work in the previous code?!?

Reply:
This has to do with the expression type derivation of the ternary operator I think, since one of the results was always a `jl_value_t*` it never resolved to `set_const<jl_value_t*&>`.